### PR TITLE
Support vector input ports in differential equations

### PR DESCRIPTION
**Pull request by pnbabu**
This PR is a continuation of #836.

### CODE REVIEW

File: doc/running.rst [hunk 1]
```diff
@@ -155,6 +155,96 @@ Custom templates
...
+
+The above code querying for ``receptor_types`` gives a list of port names and NEST ``rport`` numbers as shown below:
+
+.. list-table::
+   :header-rows: 1
+
+   * - Input port name
+     - NEST ``rport``
+   * - AMPA_spikes
+     - 0
```
Comment (clinssen) [inline doc/running.rst:224]: I'm still confused; I thought NEST started counting at 1 for "multisynapse" neurons. That's also what is described here:

https://github.com/nest/nestml/pull/849/files#diff-cca67b57fc552badda2b746e8ee77ea760772e475caa48ccc820d40d15f2ee1bR205

File: doc/nestml_language/neurons_in_nestml.rst [hunk 2]
```diff
@@ -204,10 +201,19 @@ The input ports can also be defined as vectors. For example,
          exc_spikes[3] pA <- excitatory spike
          inh_spikes[3] pA <- inhibitory spike
        end
+
+       equations:
+         kernel I_kernel1 = exp(-1/tau_syn1*t)
+         kernel I_kernel2 = exp(-1/tau_syn2*t)
+         inline I_syn_1 pA = convolve(I_kernel1, foo[1])
```
Comment (clinssen) [inline doc/nestml_language/neurons_in_nestml.rst:208]: can you use exc/inh nomenclature here?

File: pynestml/codegeneration/nest_code_generator.py [hunk 3]
```diff
@@ -619,6 +628,12 @@ def _get_neuron_model_namespace(self, neuron: ASTNeuron) -> Dict:
                     marks_delay_vars_visitor = ASTMarkDelayVarsVisitor()
                     expr_ast.accept(marks_delay_vars_visitor)
 
+                # Check if the update expressions have vector variables and update the vector parameters
```
Comment (clinssen) [inline pynestml/codegeneration/nest_code_generator.py:631]: can this be done somewhere more meaningful than the ``_get_neuron_model_namespace()`` method?

File: pynestml/utils/ast_utils.py [hunk 4]
```diff
@@ -19,9 +19,9 @@
 # You should have received a copy of the GNU General Public License
 # along with NEST.  If not, see <http://www.gnu.org/licenses/>.
 
+import re
```
Comment (clinssen) [inline pynestml/utils/ast_utils.py:22]: preferably ``typing`` is first and by itself

File: pynestml/visitors/ast_equations_with_vector_variables.py [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+#
+# NEST is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+from pynestml.meta_model.ast_ode_equation import ASTOdeEquation
+
```
Comment (clinssen) [inline pynestml/visitors/ast_equations_with_vector_variables.py:22]:
```suggestion
```

File: tests/nest_tests/nest_multisynapse_test.py [hunk 6]
```diff
@@ -112,3 +112,52 @@ def test_multisynapse(self):
...
+
+        mm = nest.Create("multimeter", params={"record_from": [
+            "I_kernel1__X__spikes1", "I_kernel2__X__spikes2", "I_kernel3__X__spikes3"], "interval": 0.1})
+        nest.Connect(mm, neuron)
+
+        vm_1 = nest.Create("voltmeter")
+        nest.Connect(vm_1, neuron)
+
+        # simulate
+
```
Comment (clinssen) [inline tests/nest_tests/nest_multisynapse_test.py:157]:
```suggestion
```

**Review by clinssen — COMMENTED** [review]
Great, thanks! Just some minor points.

**Comment by pnbabu** [comment]
The CI fails with this error:
```
* [404] [https://github.com/nest/nestml/blob/master/tests/nest_tests/resources/iaf_psc_exp_multisynapse_vectors.nestml](https://github.com/nest/nestml/blob/master/tests/nest_tests/resources/iaf_psc_exp_multisynapse_vectors.nestml) | Failed: Network error: Not Found
```
The file `iaf_psc_exp_multisynapse_vectors.nestml` is newly added in this PR, so it's not present on the master branch yet. @clinssen Should I provide a path with my branch instead of the master and then change it again to master after this PR is merged?

**Pull request merged**